### DISCUSSION TITLE
Generating polycrystalline tessellations and running CPFE models

### DISCUSSION THREAD

**Discussion by abarun22** [idaholab/moose#24881] [Q&A General]
Hi,
Is there a possibility in Moose to create polycrystaline structures in the form of tessallations and use that for crystal plasticity simulations? These tessallations would obviously need meshed in a suitable format that Moose understands. I have managed to create the polycrystalline meshes through neper and used that in Moose, but felt would be more convenient if all of them can be done within Moose. Can some one point me to the right set of examples?
Kind regards,
Arun

**Comment by GiudGiud** [comment]
Hello

I think we don’t have the meshing capability you want here, unless you are willing to define the boundaries of the grains manually. 

However we can initialize variables on a regular mesh with a grain pattern. There s the PolyCrystalColoringIC for example

Guillaume

**Comment by hugary1995** [comment]
We use neper/dream3d for tessellation then cubit/sculpt for hex mesh. What do you think would be the benefit of having moose native polycrystalline tessellation?

**Reply by sapitts** [reply]
We use this same approach as well for our crystal plasticity simulations. The cubit/sculpt step can be tricky, yet is worth the effort to avoid the "stairstepping" of the mesh at grain boundaries.

**Reply by abarun22** [reply]
I am not sure if Moose based tessellations has any potential advantages over the neper route. Also not very clear how the moose tessellations are handled further to create the mesh. Does moose output any file format that can describe tessellations and related properties?

**Reply by PengWei97** [reply]
Hi @hugary1995,
> We use neper/dream3d for tessellation then cubit/sculpt for hex mesh. What do you think would be the benefit of having moose native polycrystalline tessellation?

I also want to use neper to create polycrystalline structures for use in moose. Now what I want to ask is, does such an operation require the creation of additional objects in moose? In particular, do you need to create a UserObjects based on `PropertyReadFile` to manage the Euler angle sequence created by neper ?

**Reply by hugary1995** [reply]
You need to read in euler angles one way or the other. If there's no such userobject available in moose, you'll have to write your own.

**Reply by tanuj163** [reply]
What file did you export from DREAM3D to import in CUBIT? @hugary1995
